File: easy/subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function (n) {
  const digits = String(n)
    .split('')
    .map((n) => Number(n));
  const product = digits.reduce((p, c) => p * c);
  const sum = digits.reduce((p, c) => p + c);
  return product - sum;
};

File: easy/subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.c
int subtractProductAndSum(int n)
{
    int sum = 0;
    int product = 1;
    while (n > 0)
    {
        int num = n % 10;
        sum += num;
        product *= num;
        n /= 10;
    }
    return product - sum;
}